exe/genv-envs

```diff
@@ -130,10 +179,28 @@ def do_config(envs: Envs, eid: str, command: str, args: argparse.Namespace) -> N
     elif command == 'gpu-memory':
         gpu_memory(args.gpu_memory, args.clear)
 
+
 def do_deactivate(envs: Envs, pid: int) -> None:
+    """
+    Deactivate the genv environment with the current shell.
+    Removes the current shell connection to it's environment, and cleans all jupyter kernels (?!).
```

Comment:
actually you are right!

in general, we cleanup every time, and we do this by looking for processes and kernel (processes) that are not running anymore.
a regular `cleanup()` would also run before this `do_deactivate` is executed.

here, the process is still running, it just wants to deactivate, so the normal `poll_pid` won't work (because the process is running)

this is why we pass a custom `poll_pid` lambda.

so probably the right thing to do here is to pass `poll_kernel=None`
because here we want to deactivate only this (running) process and that's it.

this function is _not_ responsible for doing cleanups

wdyt?

Reply:
I think that you are right - we should pass `poll_kernel=None`. What we need to consider is the senario in which the genv-deactivate is called insdie a jupyter kernel (not sure that it is a problem, we just need to see that everything is working as expected).

Reply:
haha good point
but currently jupyter kernels are managed by the Python package `jupyterlab_genv` and in particular the kernel provisioner there
so generally speaking, managing genv using bash inside a Jupyter is not really supported I guess 🤷

Reply:
Yes, but in another PR. I will add it to the "to do" list.